### PR TITLE
Add GP ORTs

### PR DESCRIPTION
**Pull request by dustinswales**
# Attention

This PR was merged with the wrong fv3atm submodule pointer hash. The hash referenced in this branch (cf0a73180b2d9ac55ebfce4785a7270d205423db) resides/d in @dustinswales's fv3atm fork. The correct fv3atm hash in the NOAA-EMC repository would have been 86d4bb39cbc9591288cab3b07f9edfa4a345ee7f.

# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [ ] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

This PR contains new operational regression tests for RRTMGP in physics prototype 7.

### Issue(s) addressed

See  https://github.com/NCAR/ccpp-physics/pull/782 for description of issues.

## Testing

Testing at commit b707db5 on Cheyenne against baseline ``develop-20211203`` shows the following:

GNU: all jobs ran to completion. The following two tests failed comparison (as expected):
```
control_rrtmgp
control_rrtmgp_debug
```

INTEL: All non-RRTMGP tests ran to completion and passed. The following test ran to completion but failed comparison (as expected):
```
control_rrtmgp_debug
```

The following jobs compiled but failed at runtime:
```
cpld_control_p7_rrtmgp
control_p7_rrtmgp
regional_RRTMGP
control_rrtmgp
control_rrtmgp_c192
```

These were tested on Intel and gnu on Hera.
This PR contains new tests to be added to tests/rt.conf.

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [x] opnReqTest for newly added/changed feature
- [x] CI

## Dependencies

- waiting on https://github.com/NCAR/ccpp-physics/pull/782
- waiting on https://github.com/NOAA-EMC/fv3atm/pull/428

### CODE REVIEW

**Comment by DeniseWorthen** [comment]
Please update to the last commit for FV3 and ufs-weather.

**Comment by dustinswales** [comment]
@DeniseWorthen 
Should be all up-to-date now

File: tests/tests/control_p7_rrtmgp [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+export MIN_SEAICE=0.15
+export FRAC_GRID=.true.
+export MOM6_RESTART_SETTING=n
+
+# RRTMGP
+export DO_RRTMGP=.true.
+
+export FV3_RUN=control_run.IN
+export CCPP_SUITE=FV3_GFS_v16_p7_rrtmgp
+export NEW_DIAGTABLE=diag_table_gfsv16_merra2
```
Comment (DeniseWorthen) [inline tests/tests/control_p7_rrtmgp]: This needs to be consistent with the newest control_p7 test (NEW_DIAGTABLE-> DIAGTABLE)

Comment (dustinswales) [reply tests/tests/control_p7_rrtmgp]: Done.

**Comment by DeniseWorthen** [comment]
Could you update your PR information for the CCPP-physics PR? It is showing as closed. 

Also, is there an issue in either FV3 or CCPP that you can list in the 'issues addressed' section?

**Comment by dustinswales** [comment]
@DeniseWorthen 
I updated the ccpp-physics link in the PR.
There are no open issues to reference.

**Comment by MinsukJi-NOAA** [comment]
I am not able to build `control_p7_rrtmgp` and `cpld_control_p7_rrtmgp` with the following errors:
```
1403:/scratch2/NCEPDEV/stmp1/Minsuk.Ji/PR946_2/FV3/ccpp/data/GFS_typedefs.F90(2109): error #6459: This field name has been defined more than once within the same structure.   [SCMPSW]
1406:/scratch2/NCEPDEV/stmp1/Minsuk.Ji/PR946_2/FV3/ccpp/data/GFS_typedefs.F90(2109): error #6227: This symbol has multiple POINTER statement/attribute declarations which is not allowed.   [SCMPSW]
1409:/scratch2/NCEPDEV/stmp1/Minsuk.Ji/PR946_2/FV3/ccpp/data/GFS_typedefs.F90(2028): error #7367: The data value NULL() can only be assigned to a Fortran POINTER.
1412:/scratch2/NCEPDEV/stmp1/Minsuk.Ji/PR946_2/FV3/ccpp/data/GFS_typedefs.F90(2109): error #7367: The data value NULL() can only be assigned to a Fortran POINTER.
1415:/scratch2/NCEPDEV/stmp1/Minsuk.Ji/PR946_2/FV3/ccpp/data/GFS_typedefs.F90(7570): error #6158: The structure-name is invalid or is missing.   [INTERSTITIAL]
1419:make[2]: *** [FV3/ccpp/physics/CMakeFiles/ccpp_physics.dir/__/data/GFS_typedefs.F90.o] Error 1
1422:make[1]: *** [FV3/ccpp/physics/CMakeFiles/ccpp_physics.dir/all] Error 2
1424:make: *** [all] Error 2
```

**Comment by dustinswales** [comment]
@MinsukJi-NOAA 
Sorry, there were multiple declarations/allocations of Interstitial%scmpsw in GFS_typedefs.F90.
(Not sure why the merge didn't catch this, but should be good to go now)

**Comment by MinsukJi-NOAA** [comment]
@dustinswales Could you grab the ORT results for `control_p7_rrtmgp` (`/scratch2/NCEPDEV/stmp1/Minsuk.Ji/PR946_2/tests/OpnReqTests_hera.intel.log`) and commit? ORT for `cpld_control_p7_rrtmgp` cannot be run at this time due to wave component.

**Comment by dustinswales** [comment]
@MinsukJi-NOAA 
NP. done

**Comment by DeniseWorthen** [comment]
Please update ``BL_DATE`` in rt.sh to 20211206.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: BL
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/796090531/20211206214514/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_29447
Baseline creation and move successful
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/796090531/20211206225243/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_13620
Test hafs_global_multiple_4nests_atm 087 failed failed 
Test hafs_global_multiple_4nests_atm 087 failed in run_test failed 
Please make changes and add the following label back:
hera-intel-BL

**Comment by junwang-noaa** [comment]
hafs_global_multiple_4nests_atm test timed out on hera. I am rerunning this test.

**Comment by DeniseWorthen** [comment]
@dustinswales Please commit the wcoss logs from here: /scratch2/NCEPDEV/stmp1/Denise.Worthen/PR946

**Comment by DeniseWorthen** [comment]
I'm not sure what happened with the hera.gnu RT. The baseline is present, but I can't find any jobs in the queue and the log hasn't been posted. I'll run it manually.

**Comment by junwang-noaa** [comment]
I was wondering that too. I can't find where is the code directory that runs the hera gnu tests. 

By the way. the hera.intel log file is at:
/scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/796090531/20211206225243/ufs-weather-model1/tests/RegressionTests_hera.intel.log

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: intel
Job: BL
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/796090531/20211206194510/ufs-weather-model
Please manually delete: /glade/scratch/dtcufsrt/FV3_RT/rt_51266
Test cpld_control_p7_rrtmgp 002 failed failed 
Test cpld_control_p7_rrtmgp 002 failed in run_test failed 
Test control_p7_rrtmgp 019 failed failed 
Test control_p7_rrtmgp 019 failed in run_test failed 
Test regional_RRTMGP 024 failed failed 
Test regional_RRTMGP 024 failed in run_test failed 
Test control_rrtmgp 030 failed failed 
Test control_rrtmgp 030 failed in run_test failed 
Test control_rrtmgp_c192 031 failed failed 
Test control_rrtmgp_c192 031 failed in run_test failed 
Please make changes and add the following label back:
cheyenne-intel-BL

**Comment by junwang-noaa** [comment]
The hera gnu log file is at: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/796090531/20211206225243/gnu/ufs-weather-model/tests/RegressionTests_hera.gnu.log

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: jet
Compiler: intel
Job: BL
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/796090531/20211207023013/ufs-weather-model
Please manually delete: /lfs4/HFIP/h-nems/emc.nemspara/RT_RUNDIRS/emc.nemspara/FV3_RT/rt_25639
Test control_lndp 017 failed failed 
Test control_lndp 017 failed in run_test failed 
Please make changes and add the following label back:
jet-intel-BL

**Comment by DeniseWorthen** [comment]
@dustinswales I wanted to add the testing information using your PR branch against the current baseline to document which tests changed (see the information I added in the PR description). Jobs on Cheyenne.intel failed with the following (this is the control_rrtmgp err file). Cheyenne uses a newer intel compiler compared to the other platforms. Do you have access to Cheyenne?

```
101:MPT: #6  0x000000000369ded3 in mo_gas_optics_kernels::interpolate3d_byflav (
101:MPT:     res=..., scaling=..., fmajor=..., k=...,
101:MPT:     gpts=<error reading variable: Cannot access memory at address 0xc>,
101:MPT:     gpte=<error reading variable: Cannot access memory at address 0x0>,
101:MPT:     jeta=..., jtemp=10, jpress=2)
101:MPT:     at /glade/work/worthen/dustin_intel/FV3/ccpp/physics/physics/rte-rrtmgp/rrtmgp/kernels/mo_gas_optics_kernels.F90:729
101:MPT: #7  0x000000000369d94e in mo_gas_optics_kernels::gas_optical_depths_major (
101:MPT:     ncol=-573113600, nlay=1167938859, nbnd=-609229083, ngpt=-669452182,
101:MPT:     nflav=<error reading variable: Cannot access memory at address 0xc>,
101:MPT:     neta=<error reading variable: Cannot access memory at address 0x0>,
101:MPT:     npres=59, ntemp=14, gpoint_flavor=..., band_lims_gpt=..., kmajor=...,
101:MPT:     col_mix=..., fmajor=..., jeta=..., tropo=..., jtemp=..., jpress=...,
101:MPT:     tau=...)
```

This seems to be the same error which caused the actual BL creation on Cheyenne to fail:

```
86:MPT: #6  0x000000000369ded3 in mo_gas_optics_kernels::interpolate3d_byflav (
86:MPT:     res=<error reading variable: value requires 62545920 bytes, which is more than max-value-size>, scaling=..., fmajor=...,
86:MPT:     k=<error reading variable: value requires 1619852943 bytes, which is more than max-value-size>,
86:MPT:     gpts=<error reading variable: Cannot access memory at address 0xa>,
86:MPT:     gpte=<error reading variable: Cannot access memory at address 0x0>,
86:MPT:     jeta=..., jtemp=10, jpress=2)
86:MPT:     at /glade/scratch/dtcufsrt/autort/tests/auto/pr/796090531/20211206194510/ufs-weather-model/FV3/ccpp/physics/physics/rte-rrtmgp/rrtmgp/kernels/mo_gas_optics_kernels.F90:729
```

**Comment by junwang-noaa** [comment]
@dustinswales hera is on PM today. I put the hera and wocss log files on jet at: /lfs4/HFIP/h-nems/emc.nemspara/manualrt/logfiles

**Comment by dustinswales** [comment]
> @dustinswales hera is on PM today. I put the hera and wocss log files on jet at: /lfs4/HFIP/h-nems/emc.nemspara/manualrt/logfiles

@junwang-noaa 
Sorry, I don't have access to jet, only hera

**Comment by junwang-noaa** [comment]
@dustinswales Do you have access to Cheyenne?

**Comment by dustinswales** [comment]
@DeniseWorthen 
I just read the updated PR description. 
I have access to Cheyenne, but no project allocation to do UFS related work there.
Can you point me to you run directory on Cheyenne? Then I can see why there are errors there, but not on the NOAA machines.

**Comment by DeniseWorthen** [comment]
The run directories on Cheyenne are:

GNU:  /glade/scratch/worthen/FV3_RT/rt_8804
INTEL: /glade/scratch/worthen/FV3_RT/rt_19802

The two build directories are in /glade/work/worthen/dustin_gnu and dustin_intel. These are the runs I did using the current baseline.

**Comment by dustinswales** [comment]
@DeniseWorthen @junwang-noaa 
There are many differences int the input.nml files between the GP/G runs.
The only difference between the G/GP tests should be the extra GP nml options (https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/parm/control_rrtmgp.nml.IN#L180), but when I look at the test files it seems that they have diverged:
https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/parm/control.nml.IN
https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/parm/control_rrtmgp.nml.IN

**Comment by dustinswales** [comment]
@DeniseWorthen @junwang-noaa 
I was able to trace back to where (way back) the GP RT's got messed up.
https://github.com/ufs-community/ufs-weather-model/commit/b4746793c6b37e35c6fe728bf1e9018f6a4815ca#diff-2081da5ae6d322f7299c67fc6cea283019c8ea800a7d14d49115818de593e65b

Screen Shot 2021-12-07 at 9 11 46 AM

**Comment by junwang-noaa** [comment]
Thanks, Dustin. So the rrtmgp namelist is from previous gfs_v16_c192L127 as the PR #561 is to use 127 level ICs.  Any RRTMGP namelist setting needs to be changed for C96 resolution?

**Comment by dustinswales** [comment]
The GP RT's should be identical to the G tests with two exceptions, 1) the nml options for GP and 2) corresponding GP SDF.

**Comment by DeniseWorthen** [comment]
Does the namelist explain why this PR failed (ie, the code seg-faulted) on only one platform/compiler?

**Comment by dustinswales** [comment]
@DeniseWorthen 
I can't say for sure what's going on.
GP was passing all of the RTs (rt.sh) and ORTs (opnReqTest) a few weeks ago when I opened the PRs. 
We've never had any issues in the past running the basic GP RTs (control_rrtmgp, control_rrtmgp_debug, regional_RRTMGP, control_rrtmgp_2threads) on Cheyenne. 

A few comments.
- The GP RTs need to be updated to be consistent with the RRTMG tests, especially if opnReqTest is not working currently with WAVES. 
- If opnReqTest is working and a requirement for inclusion, then maybe we don't need all of these GP RTs (e.g. control_rrtmgp_debug/2threads/etc)? But maybe since the RTs and ORTs use different SDF this is not the best?

I'm happy to help getting the GP RTs working, but it will have to be when Hera comes back online.

**Comment by DeniseWorthen** [comment]
On Cheyenne, the control_p7_rrtmgp test fails w/ the same error as the control_rrtmgp test. The _p7 test uses the identical input nml to control_p7, differing only in the SDF and the rrtmgp namelist variables. 

We also have run all previous commits on Cheyenne.intel which included the existing rrtmgp tests. The comparison I was doing was against the 1203 baseline which included these same tests. I did not expect your PR to compare B4B w/ the previous baseline. But this PR is failing at runtime.

**Comment by dustinswales** [comment]
> On Cheyenne, the control_p7_rrtmgp test fails w/ the same error as the control_rrtmgp test. The _p7 test uses the identical input nml to control_p7, differing only in the SDF and the rrtmgp namelist variables.

Good point.
The errors in your run directory (/glade/scratch/worthen/FV3_RT/rt_19802/control_p7_rrtmgp) point to corrupt data in the rte-rrtmgp code:
MPT: #6  0x0000000003798143 in mo_gas_optics_kernels::interpolate3d_byflav (
MPT:     res=<error reading variable: value requires 4024098816 bytes, which is more than max-value-size>, scaling=..., fmajor=..., k=...,
MPT:     gpts=<error reading variable: Cannot access memory at address 0xa>,
MPT:     gpte=<error reading variable: Cannot access memory at address 0x0>,
MPT:     jeta=..., jtemp=9, jpress=2)
MPT:     at /glade/work/worthen/dustin_intel/FV3/ccpp/physics/physics/rte-rrtmgp/rrtmgp/kernels/mo_gas_optics_kernels.F90:729
MPT: #7  0x0000000003797bbe in mo_gas_optics_kernels::gas_optical_depths_major (
MPT:     **ncol=355571584, nlay=1190934089, nbnd=-258150488, ngpt=808798886,**

But no errors in the .out file. nml being read-in appears correct. GP is well passed initialization at this point of failure.
Which version of Intel is being used on Cheyenne?
There are some changes to the rte-rrtmgp code that I can't speak to. 
@RobertPincus?

> 
> We also have run all previous commits on Cheyenne.intel which included the existing rrtmgp tests. The comparison I was doing was against the 1203 baseline which included these same tests. I did not expect your PR to compare B4B w/ the previous baseline. But this PR is failing at runtime.

I don't have an answer for this.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: jet
Compiler: intel
Job: RT
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/796090531/20211207150016/ufs-weather-model
Please manually delete: /lfs4/HFIP/h-nems/emc.nemspara/RT_RUNDIRS/emc.nemspara/FV3_RT/rt_301487
Test control_csawmg_debug 063 failed in run_test failed 
Please make changes and add the following label back:
jet-intel-RT

**Comment by RobertPincus** [comment]
@dustinswales @BrianCurtis-NOAA @DeniseWorthen I'm not sure what to make of this. The same code is running in simulations in other models (e.g. ICON). As Dustin notes, the values of variables being passed to low-level functions are clearly nutty. These values are being set in higher level interface routines. Nor do I understand the line 
MPT: res=<error reading variable: value requires 4024098816 bytes, which is more than max-value-size>
There aren't any arrays anywhere near 4Gb in RRTMGP.

**Comment by DeniseWorthen** [comment]
Regarding the Jet-intel failure, the ``control_csawmg_debug`` test is failing comparison of the ``atmf000.nc`` file. The script-level comparison then stops and reports the test failed. If I do a manual comparison of the ``atmf001.nc`` file, it passes. Both ``sfcf`` files (000,001) pass comparison.

**Comment by junwang-noaa** [comment]
@RobertPincus Thanks for checking, since the control_RRTMGP test was working in the current code on Cheyenne, we may need to take a look at the new code changes, additional updates may be needed for those code changes to run with Intel 2021.2 version as this is the compiler on Cheyenne. Do you have access to Cheyenne?

**Comment by RobertPincus** [comment]
@junwang-noaa I do not have access to Cheyenne I'm afraid. You are welcome to look at the changes to RRTMGP but note that standalone tests are run as part of continuous integration and are succeeding with ifort 2011.04 (see https://github.com/earth-system-radiation/rte-rrtmgp/runs/4415129750?check_suite_focus=true)

**Comment by climbfuji** [comment]
The weird thing is that the same setup runs in DEBUG mode, but not in PROD or REPRO mode. I'll use the parallel debugger tomorrow morning to see why it is failing in that place.

**Comment by climbfuji** [comment]
> The weird thing is that the same setup runs in DEBUG mode, but not in PROD or REPRO mode. I'll use the parallel debugger tomorrow morning to see why it is failing in that place.
> […](#)

When I run this through allinea ddt, I get weird values for gptS, see attached screenshot.

Screen Shot 2021-12-08 at 6 38 14 AM

**Comment by RobertPincus** [comment]
@DomHeinzeller That is certainly an error. It suggests corrupted memory, I guess.

**Comment by BrianCurtis-NOAA** [comment]
@DomHeinzeller is `-fsanitize=address` set in the compiler options? that may help? Although I've seen weird integers like that even when the build passes with sanitize.

**Comment by climbfuji** [comment]
> @DomHeinzeller is `-fsanitize=address` set in the compiler options? that may help? Although I've seen weird integers like that even when the build passes with sanitize.

We don't have this compiler flag yet, will try manually.

This is the screenshot for exactly the same run, same place, when compiled in DEBUG mode. The values look good.

Screen Shot 2021-12-08 at 7 34 35 AM

**Comment by DeniseWorthen** [comment]
@dustinswales would you please merge Dom's PR to update the rt.conf? Thanks.

**Comment by climbfuji** [comment]
@dustinswales I have a little more info. The repro mode is apparently broken due to previous build system updates (recall that we don't really test the repro mode ... a flaw). I run in prod mode, I get more information on the variables. I stopped the model in DDT before the offending line. The screenshot shows that jeta(2) has one bogus (for reasons I do not know yet).

Screen Shot 2021-12-08 at 10 10 32 AM

Also look at variables fmajor and k ...

Screen Shot 2021-12-08 at 10 15 04 AM

**Comment by RobertPincus** [comment]
@DomHeinzeller @dustinswales Variable `jeta` is set on line 104 of the same file. Tracing up you can see there should be no way to get a negative value. I don't know what to make of the debugger not being able to read `fmajor`

**Comment by DomHeinzeller** [comment]
This probably happens because something else is writing into the variables ... I guess

**Comment by RobertPincus** [comment]
I guess too, but who? And why would someone do such a thing 😢

**Comment by RobertPincus** [comment]
@DomHeinzeller In more seriousness, these variables are all internal to RRTMGP. The big tables like `kmajor` are private components of class structures. The smaller variables like `jeta` are internal to the subroutines.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: jet
Compiler: intel
Job: RT
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/796090531/20211208161511/ufs-weather-model
Please manually delete: /lfs4/HFIP/h-nems/emc.nemspara/RT_RUNDIRS/emc.nemspara/FV3_RT/rt_187200
Test control_stochy 026 failed in check_result failed 
Test control_stochy 026 failed in run_test failed 
Please make changes and add the following label back:
jet-intel-RT

**Comment by DeniseWorthen** [comment]
@dustinswales Logs are not being posted from any of the RTs because they can't get pushed to your fork. I'll collect them in once place for you to commit. 

The cheyenne.intel baseline created but I think I'm not sure the verify ran or where it is located. I may just run it manually if I can't find it.

**Comment by DeniseWorthen** [comment]
The jet.intel failure of the ``control_stochy`` test was another case of the atmf000.nc file not comparing (same file as the ``control_csawmg_debug`` failure seen earlier).

**Comment by DeniseWorthen** [comment]
@dustinswales Please commit the logs in /scratch2/NCEPDEV/stmp1/Denise.Worthen/PR946

**Review by junwang-noaa — APPROVED** [review]

**Review by MinsukJi-NOAA — APPROVED** [review]

**Comment by DusanJovic-NOAA** [comment]
I am getting build error on my laptop when compiling mo_gas_optics_kernel:

```
[1/1075] Generating Fortran dyndep file FV3/ccpp/physics/CMakeFiles/ccpp_physics.dir/Fortran.dd
ninja: build stopped: multiple rules generate FV3/ccpp/physics/mo_gas_optics_kernels.mod.
```
in CMakeLists.txt I see two `mo_gas_optics_kernels.F90` being compiled, both define same module (mo_gas_optics_kernels)

```
$ grep mo_gas_optics_kernels FV3/ccpp/physics/CMakeLists.txt 
                       ${LOCAL_CURRENT_SOURCE_DIR}/physics/rte-rrtmgp/rrtmgp/kernels-openacc/mo_gas_optics_kernels.F90
                       ${LOCAL_CURRENT_SOURCE_DIR}/physics/rte-rrtmgp/rrtmgp/kernels/mo_gas_optics_kernels.F90
```
Could this be the reason for the crash on Cheyenne.

**Comment by climbfuji** [comment]
> I am getting build error on my laptop when compiling mo_gas_optics_kernel:
> 
> ```
> [1/1075] Generating Fortran dyndep file FV3/ccpp/physics/CMakeFiles/ccpp_physics.dir/Fortran.dd
> ninja: build stopped: multiple rules generate FV3/ccpp/physics/mo_gas_optics_kernels.mod.
> ```
> 
> in CMakeLists.txt I see two `mo_gas_optics_kernels.F90` being compiled, both define same module (mo_gas_optics_kernels)
> 
> ```
> $ grep mo_gas_optics_kernels FV3/ccpp/physics/CMakeLists.txt 
>                        ${LOCAL_CURRENT_SOURCE_DIR}/physics/rte-rrtmgp/rrtmgp/kernels-openacc/mo_gas_optics_kernels.F90
>                        ${LOCAL_CURRENT_SOURCE_DIR}/physics/rte-rrtmgp/rrtmgp/kernels/mo_gas_optics_kernels.F90
> ```
> 
> Could this be the reason for the crash on Cheyenne.

Definitely, this could be the problem. Need to figure out why this happens. Thanks for pointing this out!

**Pull request merged**